src/middlewares/Fetch.ts

```diff
@@ -1,7 +1,21 @@
-import {IRequest, IResponse} from "../Client";
+import {IRequest, IResponse, IResponseHeaders} from "../Client";
 import {combineUrlWithBaseUrl, objectToQueryString} from "../internal/formatting";
 import {IMiddleware} from "../Stack";
 
+export type FetchResponse = {
+    new(body: any, init: any): FetchResponse;
+    body: string;
+    bodyUsed: boolean;
+    headers: IResponseHeaders
```

Comment:
Uhm when refactoring this to FetchResponse. I assume there will be no more IRequest Objects be present... EVER :P 

TLDR: Remove the IResponse interface. All occurrences should be replaced by FetchResponse. Same for the IRequestHeaders.